docs/NUnit-glue-library.md

```diff
@@ -0,0 +1,275 @@
+
+## `Frozen` attribute
+
+As shown above, the `AutoData` and `InlineAutoData` can be delegated the creation of the system under test. Doing so creates an interesting challenge if the system under test accepts dependencies that need to be configured or used during the _Assert_ phase.
+
+Let's take the following class as test subject and assume we want to rewrite the unit test using the `AutoData` attribute.
+```csharp
+public class Service
+{
+    private readonly IList<string> _items;
+    
+    public Service(IList<string> items)
+    {
+        _items = items ?? throw new ArgumentNullException(nameof(items));
+    }
+    
+    public void Add(string item) => _items.Add(item ?? throw new ArgumentNullException(nameof(item)));
+}
+
+[Test]
+public void Add_should_add_item_to_underlying_list()
+{
+    // ARRANGE
+    var fixture = new Fixture();
+    var list = fixture.Create<List<string>>();
+    var sut = new Service(list);
+    var item = fixture.Create<string>();
+
+    // ACT
+    sut.Add(item);
+
+    // ASSERT
+    Assert.That(list, Contains.Item(item));
+}
+```
+
+Unfortunately, simply decorating the test with the `AutoData` attribute and convert the variables `list`, `sut` and `item` as parameters will not work because the list served as parameter and the one served to the constructor of `Service` will not be the same instance.
+
+The `Frozen` article solves this issue. By leveraging the [`Freeze`](Type-customization#freeze) extension method, it creates an instance of a given type and it uses it to serve successive requests. This attribute is used by decorating which parameter of the unit test needs to be generated using the `Freeze` method and it works with test decorated by both the `AutoData` and `InlineAutoData` attributes.
+
+With the help of the `Frozen` attribute, we can rewrite the test as follow
+
+```csharp
+[Test, AutoData]
+public void Add_should_add_item_to_underlying_list([Frozen] IList<string> list, Service sut, string item)
+{
+    // ACT
+    sut.Add(item);
+
+    // ASSERT
+    Assert.That(list, Contains.Item(item));
+}
+```
+
+Please notice the sequence of parameters of the rewritten unit test: the frozen parameters must go before the class using them.
+
+## `Greedy` and `Modest` attributes
+
+Another issue deriving from delegating AutoFixture of instantiating classes like system under tests is the loss of control on which constructor is picked. By default, AutoFixture picks the constructor with least parameters, but sometimes this is not the optimal choice.
+
+The `Greedy` and `Modest` attributes give the developer the power to instruct AutoFixture which constructor to select when constructing an object to be passed to NUnit. The `Greedy` attribute will instruct AutoFixture to use the constructor with the most parameters while the `Modest` attribute will instruct AutoFixture in the opposite way.
```

Comment:
> in the opposite way

The opposite way how?
If it's just picking the constructor with the least parameters then why would someone use it since it's the default behavior? 

Could we also add a short sentence about the scenario when the constructor you want to pick is neither the one with the most or the least parameters.

Reply:
Honestly, the need for `[Modest]` is unclear to me too.